### PR TITLE
Add SignalR Daily Tests scripts

### PR DESCRIPTION
**Pull request by analogrelay**
This adds a YAML build script to run SignalR "Daily Tests". Right now, that just enables running browser tests in Sauce Labs instead of locally.

I'm still messing with the build definition. @natemcmaster take a look and see what you think, feel free to tear it apart ;P. The goal is simple: Run the standard build script, over the `src/SignalR` directory but with the additional MSBuild property `DailyTests=true`. That's all we need to trigger the daily tests.

Things to be done after the merge:
* [x] Change the [build definition](https://dev.azure.com/dnceng/internal/_build?definitionId=132&_a=summary) to run on `master` and set up the schedule
* [x] Set up notifications to the SignalR team on failures
* [ ] Decide on a way to manage flaky tests. I've noticed a few timeouts creep in that I haven't been able to diagnose. We should try to avoid getting into a situation where we just ignore failures here though. Perhaps have something that detects timeouts and leaves them, or rerun the tests, or increase timeouts, etc.

### CODE REVIEW

File: src/SignalR/clients/ts/FunctionalTests/scripts/run-tests.ts [hunk 1]
```diff
@@ -199,7 +242,7 @@ function runJest(httpsUrl: string, httpUrl: string) {
         if (sauce) {
             // SauceLabs can only proxy certain ports for Edge and Safari.
             // https://wiki.saucelabs.com/display/DOCS/Sauce+Connect+Proxy+FAQS
-            desiredServerUrl = "http://127.0.0.1:9000";
+            desiredServerUrl = "http://127.0.0.1:9000;https://127.0.0.1:9001";// Sauce Labs can only proxy certain
```
Comment (BrennanConroy) [inline src/SignalR/clients/ts/FunctionalTests/scripts/run-tests.ts]: fix up comment

File: src/SignalR/clients/ts/FunctionalTests/scripts/run-tests.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { ChildProcess, exec, spawn } from "child_process";
 import { EOL } from "os";
 import { Readable } from "stream";
 
+import * as os from "os";
```
Comment (BrennanConroy) [inline src/SignalR/clients/ts/FunctionalTests/scripts/run-tests.ts]: Is this used or just an artifact of making these changes?

File: src/SignalR/clients/ts/FunctionalTests/scripts/run-tests.ts [hunk 3]
```diff
@@ -243,14 +302,48 @@ function runJest(httpsUrl: string, httpUrl: string) {
...
+            httpsUrl = httpsUrl.replace(/localhost/g, hostname);
+            httpUrl = httpUrl.replace(/\d+\.\d+\.\d+\.\d+/g, hostname);
+            httpsUrl = httpsUrl.replace(/\d+\.\d+\.\d+\.\d+/g, hostname);
+        }
+
+        conf.client.args = [];
+
+        if (sauce) {
+            // Configure Sauce Connect logging
+            conf.sauceLabs.connectOptions.logfile = path.resolve(LOGS_DIR, "sc.log");
```
Comment (BrennanConroy) [inline src/SignalR/clients/ts/FunctionalTests/scripts/run-tests.ts]: "sauceConnect.log"?

File: src/SignalR/clients/ts/FunctionalTests/ts/Common.ts [hunk 4]
```diff
@@ -4,40 +4,55 @@
...
     ENDPOINT_BASE_URL = httpServer;
     ENDPOINT_BASE_HTTPS_URL = httpsServer;
-    console.log(`Using SignalR Server: ${ENDPOINT_BASE_URL}`);
 } else if (typeof document !== "undefined") {
     ENDPOINT_BASE_URL = `${document.location.protocol}//${document.location.host}`;
 } else if (process && process.env && process.env.SERVER_URL) {
     const urls = process.env.SERVER_URL.split(";");
-    ENDPOINT_BASE_HTTPS_URL = urls[0];
     ENDPOINT_BASE_URL = urls[1];
+    ENDPOINT_BASE_HTTPS_URL = urls[0];
```
Comment (BrennanConroy) [inline src/SignalR/clients/ts/FunctionalTests/ts/Common.ts]: This should be `urls[1]` now, and the one above should be `0`

File: src/SignalR/clients/ts/FunctionalTests/scripts/run-tests.ts (hunk 2 above)
Comment (anurse) [reply src/SignalR/clients/ts/FunctionalTests/scripts/run-tests.ts]: Old code. Will remove

File: src/SignalR/clients/ts/FunctionalTests/ts/Common.ts (hunk 4 above)
Comment (anurse) [reply src/SignalR/clients/ts/FunctionalTests/ts/Common.ts]: Actually no, the order is "right" but not consistent. The node invocation uses a different order. I'll make them both consistent. HTTPS is first, HTTP is second.

**Comment by anurse** [comment]
The updated YAML scripts are now running properly in the mondo repo: https://dev.azure.com/dnceng/internal/_build?definitionId=132&_a=summary

**Review by BrennanConroy — APPROVED** [review]
I'm happy with the changes

**Review by natemcmaster — APPROVED** [review]

**Pull request merged**